src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1600,6 +1600,14 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
 
         return "";
     }
+    /**
+     * Validates if the provided limit or offset is correct
+     * @param value - indicates the value provided for limit or offset
+     */
+    protected validateLimitAndOffset(value: number | undefined): boolean {
+        if (value === undefined || value < 0) return false;
```

Comment:
maybe also `|| value === null`? Or simply `value == undefined` instead of `value === undefined || value === null` (if the code style permits it)

Reply:
also: this can be simplified to `return !(value == undefined || value < 0)`;

Reply:
`qb.limit(null)` will give compile time error by typescript bcz the functions - limit and offset accepts argument of type `number | undefined`. Anything other than `number | undefined` will not be accepted.

Reply:
only if `strictNullChecks` is enabled, which is not by default.

Reply:
https://github.com/krisalay/typeorm/blob/1a610a323ac2bd28e3ca188ecfa1a04c509f34e2/tsconfig.json#L20

Reply:
... in the projects that are using TypeORM

Reply:
Thank you for pointing that issue.